### PR TITLE
Fix indentation for deeply nested menu groups

### PR DESCRIPTION
**Pull request by caseyWebb**
# :wrench: Modifying a component

## Context

This PR fixes the indentation level for deeply nested entries (entryWithChildren->entryWithChildren->entry) in Nri.Ui.SideNav.

## :framed_picture: What does this change look like?

__Before__

Screenshot 2024-02-28 at 13 57 37

__After__

Screenshot 2024-02-28 at 13 48 13

## Component completion checklist

- [x] I've gone through the relevant sections of the [Development Accessibility guide](https://paper.dropbox.com/doc/Accessibility-guide-4-Development--BiIVdijSaoijjOuhz3iTCJJ1Ag-rGoHpC91pFg3zTrYpvOCQ) with the changes I made to this component in mind
- [x] Changes are clearly documented
  - [x] Component docs include a changelog
  - [x] Any new exposed functions or properties have docs
- [x] Changes extend to the Component Catalog
  - [x] The Component Catalog is updated to use the newest version, if appropriate
  - [x] The Component Catalog example version number is updated, if appropriate
  - [x] Any new customizations are available from the Component Catalog
  - [x] The component example still has:
    - an accurate preview
    - valid sample code
    - correct keyboard behavior
    - correct and comprehensive guidance around how to use the component
- [x] Changes to the component are tested/the new version of the component is tested
- [x] Component API follows standard patterns in noredink-ui
  - e.g., as a dev, I can conveniently add an `nriDescription`
  - and adding a new feature to the component will _not_ require major API changes to the component
- [ ] If this is a new major version of the component, our team has stories created to upgrade all instances of the old component. Here are links to the stories:
  - add your story links here OR just write this is not a new major version
- [ ] Please assign the following reviewers:
  - [ ] Someone from your team who can review your PR in full and review requirements from your team's perspective.
  - [ ] Component library owner - Someone from this group will review your PR for accessibility and adherence to component library foundations.
  - [ ] If there are user-facing changes, a designer. (You may want to direct your designer to the [deploy preview](https://github.com/NoRedInk/noredink-ui#reviews--preview-environments) for easy review):
    - For writing-related component changes, add Stacey (staceyadams)
    - For quiz engine-related components, add Ravi (ravi-morbia)
    - For a11y-related changes to general components, add Ben (bendansby)
    - For general component-related changes or if you’re not sure about something, add the Design group (NoRedInk/design)

### CODE REVIEW

**Comment by bendansby** [comment]
What is being fixed here?

File: src/Nri/Ui/SideNav/V5.elm [hunk 1]
```diff
@@ -476,7 +476,7 @@ viewSidebarEntry config extraStyles entry_ =
                         [ Attributes.css
                             ([ listStyle none
                              , padding zero
-                             , margin zero
+                             , marginLeft zero |> Css.important
```
[the remark this answers is not in the document]
Comment (caseyWebb) [reply src/Nri/Ui/SideNav/V5.elm]: This function is recursive, with the call on line 486 passing `marginLeft (px 20)` as `extraStyles` (appended to this list here). Removing the `marginLeft` there breaks all indentation for groups. That said, I'm not actually sure that `extraStyles` should be applied to the `ul` at all (it's already applied to the parent `li` that contains the group. Removing `++ extraStyles` doesn't seem to impact the examples in the component catalog.

**Comment by bcardiff** [comment]
Can we add more levels in the Component Catalog to have check going forward for this styles?

**Comment by bendansby** [comment]
Good call, it'd be nice for it to be on the page for Percy to pick up.

**Comment by caseyWebb** [comment]
Screenshot 2024-02-28 at 14 22 42

Good to add here, or should we have another independent example for deeper nesting?

**Comment by bendansby** [comment]
Adding what you have in the description as a separate example wouldn't hurt.

**Review by bcardiff — APPROVED** [review]

**Pull request merged**